### PR TITLE
V2.0 add unit tests for `add_n` and `add_nevent` to `tbl_uvregression()` 

### PR DESCRIPTION
**Pull request by zdz2101**
**What changes are proposed in this pull request?**
Added unit tests to `add_n.tbl_uvregression()` and `add_nevent.tbl_uvregression()`

**If there is an GitHub issue associated with this pull request, please provide link.**

Closes https://github.com/insightsengineering/sme-tasks/issues/685

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [x] Ensure all package dependencies are installed by running `renv::install()`
- [x] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [x] If an update was made to `tbl_summary()`, was the same change implemented for `tbl_svysummary()`?
- [x] If a new function was added, function included in `_pkgdown.yml`
- [x] If a bug was fixed, a unit test was added for the bug check
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Code coverage is suitable for any new functions/features. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Begin in a fresh R session without any packages loaded. 
- [x] R CMD Check runs without errors, warnings, and notes
- [x] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [x] Update `NEWS.md` with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [x] Increment the version number using `usethis::use_version(which = "dev")` 
- [x] Run `codemetar::write_codemeta()`
- [x] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: tests/testthat/test-add_nevent.tbl_uvregression.R [hunk 1]
```diff
@@ -0,0 +1,64 @@
+skip_if_not(is_pkg_installed(c("broom.helpers", "broom", "survival", "survey"), reference_pkg = "gtsummary"))
```
Comment (ddsjoberg) [inline tests/testthat/test-add_nevent.tbl_uvregression.R:1]: Are there any tests that need survival and survey pkgs?

File: tests/testthat/test-add_n.tbl_uvregression.R [hunk 2]
```diff
@@ -0,0 +1,70 @@
+skip_if_not(is_pkg_installed(c("broom.helpers", "broom", "survival", "survey"), reference_pkg = "gtsummary"))
```
Comment (ddsjoberg) [inline tests/testthat/test-add_n.tbl_uvregression.R:1]: Are there any tests that need survival and survey pkgs?

Comment (ddsjoberg) [inline tests/testthat/test-add_n.tbl_uvregression.R:1]: I _think_ we can test the erorr messaging using a mice model. Do you want to add a test for this?

**Review by ddsjoberg — CHANGES_REQUESTED** [review]
Thank you @zdz2101 ! Just a few changes, please :) 🥇

File: tests/testthat/test-add_nevent.tbl_uvregression.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (zdz2101) [reply tests/testthat/test-add_nevent.tbl_uvregression.R]: added

Comment (zdz2101) [reply tests/testthat/test-add_nevent.tbl_uvregression.R]: i suppose not

File: tests/testthat/test-add_n.tbl_uvregression.R (hunk 2 above)
Comment (zdz2101) [reply tests/testthat/test-add_n.tbl_uvregression.R]: i suppose not

File: tests/testthat/test-add_n.tbl_uvregression.R [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      trial,
+      x = trt,
+      include = c(marker, age),
+      show_single_row = trt,
+      method = lm,
+      hide_n = TRUE
+    )
+  )
+  # total N added to table is accurate
+  expect_error(
```
Comment (ddsjoberg) [inline tests/testthat/test-add_n.tbl_uvregression.R:15]: We can use `expect_silent()` here to ensure there are no unwanted errors, warnings, or messages.

File: tests/testthat/test-add_n.tbl_uvregression.R [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        y = response,
+        method = glm,
+        method.args = list(family = binomial),
+        include = c(trt),
+        exponentiate = TRUE,
+        hide_n = TRUE
+      )
+  )
+  # N for levels added to table is accurate
+  expect_error(
```
Comment (ddsjoberg) [inline tests/testthat/test-add_n.tbl_uvregression.R:42]: We can use `expect_silent()` here to ensure there are no unwanted errors, warnings, or messages.

File: tests/testthat/test-add_nevent.tbl_uvregression.R [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        y = response,
+        method = glm,
+        method.args = list(family = binomial),
+        include = c(trt),
+        exponentiate = TRUE,
+        hide_n = TRUE
+      )
+  )
+  # total N added to table is accurate
+  expect_error(
```
Comment (ddsjoberg) [inline tests/testthat/test-add_nevent.tbl_uvregression.R:17]: We can use `expect_silent()` here to ensure there are no unwanted errors, warnings, or messages.

File: tests/testthat/test-add_nevent.tbl_uvregression.R [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      na.omit() |>
+      unique(),
+    tidyr::drop_na(trial, response, trt) |>
+      dplyr::filter(response == 1) |>
+      nrow() |>
+      as.character()
+  )
+
+  # N added to levels is accurate
+  expect_error(
```
Comment (ddsjoberg) [inline tests/testthat/test-add_nevent.tbl_uvregression.R:32]: We can use expect_silent() here to ensure there are no unwanted errors, warnings, or messages.

File: tests/testthat/test-add_nevent.tbl_uvregression.R [hunk 7]
```diff
@@ -0,0 +1,80 @@
+skip_if_not(is_pkg_installed(c("broom.helpers", "broom"), reference_pkg = "gtsummary"))
+
+test_that("add_nevent.tbl_regression() works", {
```
Comment (ddsjoberg) [inline tests/testthat/test-add_nevent.tbl_uvregression.R:3]: Can you update the the function name to add_nevent.tbl_uvregression()?

File: tests/testthat/test-add_nevent.tbl_uvregression.R [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+      dplyr::filter(response == 1) |>
+      with(table(trt)) |>
+      as.integer() %>%
+      {c(sum(.), .)} |> # styler: off
+      as.character(),
+    ignore_attr = TRUE
+  )
+})
+
+test_that("add_nevent.tbl_regression() messaging for linear model", {
```
Comment (ddsjoberg) [inline tests/testthat/test-add_nevent.tbl_uvregression.R:65]: Can you update the the function name to add_nevent.tbl_uvregression()?

**Review by ddsjoberg — CHANGES_REQUESTED** [review]
This is great @zdz2101 !! Just a couple small updates, and we can merge!

**Review by ddsjoberg — APPROVED** [review]
Wonderful, thank you @zdz2101 !!

**Pull request merged**